source/common/config/protobuf_link_hacks.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "envoy/service/discovery/v2/ads.pb.h"
+#include "envoy/service/ratelimit/v2/rls.pb.h"
+
+namespace Envoy {
+
+// Hack to force linking of the service: https://github.com/google/protobuf/issues/4221.
+// This file should be included ONLY if this hack is required.
+envoy::service::discovery::v2::AdsDummy dummy;
```

Comment:
Nit: call this `_als_dummy` or something to avoid over shadowing other dummy vars in files that this gets included into (same scheme for `rls_dummy`).

Reply:
Also make this `const` for sanity.